src/FAudio_gstreamer.c

```diff
@@ -102,19 +113,38 @@ static size_t FAudio_GSTREAMER_FillConvertCache(
 		voice->src.format->nBlockAlign,
 		NULL
 	);
-	g_assert(gst_buffer_fill(
-		src,
-		0,
-		buffer->pAudioData + (
-			voice->src.format->nBlockAlign *
-			voice->src.gstreamer->curBlock
-		),
-		voice->src.format->nBlockAlign
-	) == voice->src.format->nBlockAlign);
-	g_assert(gst_pad_push(
-		voice->src.gstreamer->srcpad,
-		src
-	) == GST_FLOW_OK);
+
+	if (	gst_buffer_fill(
+			src,
+			0,
+			buffer->pAudioData + (
+				voice->src.format->nBlockAlign *
+				voice->src.gstreamer->curBlock
+			),
+			voice->src.format->nBlockAlign
+		) != voice->src.format->nBlockAlign	)
+	{
+		LOG_ERROR(
+			voice->audio,
+			"for voice %p, failed to copy whole chunk into buffer",
+			voice
+		);
+		gst_buffer_unref(src);
+		return (size_t) -1;
+	}
+
+	if (	gst_pad_push(
+			voice->src.gstreamer->srcpad,
+			src
+		) != GST_FLOW_OK	)
```

Comment:
`GST_FLOW_EOS` might be a case you don't want to consider an error (end of stream), if this can ever happen in your situation.